src/bigquery.ts

```diff
@@ -2217,15 +2217,10 @@ export class BigQueryTimestamp {
       if (/^\d{4}-\d{1,2}-\d{1,2}/.test(value)) {
         pd = new PreciseDate(value);
       } else {
-        const floatValue = Number.parseFloat(value);
-        if (!Number.isNaN(floatValue)) {
-          pd = this.fromFloatValue_(floatValue);
-        } else {
-          pd = new PreciseDate(value);
-        }
+        pd = new PreciseDate(BigInt(value) * BigInt(1000));
       }
     } else {
-      pd = this.fromFloatValue_(value);
+      pd = new PreciseDate(value);
```

Comment:
This seems odd to me.  It looks like precisedate expects nanos, but if we're dealing with a timestamp integer from the service it would still be micros.  Isn't some conversion still needed here?

Is it worth having a test with a small timestamp value to exercise this?